src/gluonnlp/model/bert.py

```diff
@@ -385,8 +389,12 @@ def forward(self, inputs, token_types, valid_length=None, masked_positions=None)
         This is used in training or fine-tuning a BERT model.
         """
         outputs = []
-        seq_out, _ = self._encode_sequence(inputs, token_types, valid_length)
+        seq_out, attention_out = self._encode_sequence(inputs, token_types, valid_length)
```

Comment:
Here the `attention_out` can be used to get the attention of each layer in the BERT model, but `seq_out` is still the encoding result of the last layer. Do we need to obtain the encoding results of each layer in addition to the attention of each layer?
If we need to get it, I think we should also modify the snippets.
https://github.com/dmlc/gluon-nlp/blob/master/src/gluonnlp/model/transformer.py#L480-L484

```python
additional_outputs = []
total_outputs = []
for cell in self.transformer_cells:
    outputs, attention_weights = cell(inputs, mask)
    inputs = outputs
    total_output.append(outputs)
    if self._output_attention:
        additional_outputs.append(attention_weights)
```

Reply:
Yes, @kenjewu, you are right. There was a confusion: I needed attention weights to be returned for the visualization, where @eric-haibin-lin was thinking about returning encodings of all cells. So, in the end I ended up what I needed in the way Haibin told me, but that wasn't what he expected :) 

Now it is sorted out: I added new flag `output_all_encodings` that will change the resulting tensor to List containing outputs of all Encoder cells. I added few test to ensure that it works well with `output_attention` and all other flags enabled.